lib/galaxy/datatypes/text.py

```diff
@@ -1093,6 +1093,203 @@ def _looks_like_yaml(self, file_prefix: FilePrefix):
             return False
 
 
+@build_sniff_from_prefix
+class BCSLmodel(Text):
+    """BioChemical Space Language model file"""
+
+    file_ext = "bcsl.model"
+
+    def sniff_prefix(self, file_prefix: FilePrefix):
+        """
+        Determines whether the file is in .bcsl.model format
+        """
+        content = open(file_prefix.filename, "r").read()
+        keywords = ["#! rules", "#! inits", "#! definitions"]
+        return any(keyword in content for keyword in keywords)
```

Comment:
```suggestion
        keywords = ["#! rules", "#! inits", "#! definitions"]
        return any(keyword in file_prefix.contents_header for keyword in keywords)
```

Reply:
Is there a specific order in which the keywords have to appear ? Do they need to appear at the start of a line ? Anything to make this more specific would be much appreciated.

Reply:
There is no strict order given, they kind of identify individual sections of the model, but these can be in arbitrary order.
But you are right, they should appear at the start of a line, I changed this in [769bbcd](https://github.com/galaxyproject/galaxy/pull/14362/commits/769bbcdbfda98f506f69f2efa76537d4b85ab12b).